### PR TITLE
Add timer for stats notification service and improve logging

### LINKED ISSUE #720: server stops sending  periodic notifications for commit log stats
- Secondary server stops sending periodic stats notifications
   - When the server gets into this state it does respond to other verbs (Ex: scan), however, it is just stuck sending the stats notification
   - What is also observed is that server somehow recovers from this state automatically after a very long time

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
1. Replace the `while(true)` with `Timer` in StatsNotitifcationService.
2. Refactor StatsNotificationServer to accept currentAtSign and AtCommitLog to make code unit testable.
3. Remove the unnecessary async-awaits.

**- How to verify it**
1. The StatsNotificationService should continue to write the latest server commitId to monitor connections

**- Description for the changelog**

### CODE REVIEW

File: at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart [hunk 1]
```diff
@@ -39,70 +39,75 @@ class StatsNotificationService {
...
-  Future<void> writeStatsToMonitor(
-      {String? latestCommitID, String? operationType}) async {
+  void writeStatsToMonitor({String? latestCommitID, String? operationType}) {
     try {
-      latestCommitID ??= _atCommitLog!.lastCommittedSequenceNumber().toString();
+      latestCommitID ??= atCommitLog!.lastCommittedSequenceNumber().toString();
       // Gets the list of active connections.
-      var connectionsList =
-          InboundConnectionPool.getInstance().getConnections();
+      connectionsList ??= InboundConnectionPool.getInstance().getConnections();
```
Comment (VJag) [inline at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart]: Can "connectionsList " be a local variable? The way it was before.

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart [hunk 2]
```diff
@@ -310,10 +310,10 @@ class AtSecondaryServerImpl implements AtSecondaryServer {
     try {
       command = SecondaryUtil.convertCommand(command);
       await executor!.execute(command, connection, verbManager!);
-    } on Exception catch (e, trace) {
+    } on Exception catch (e) {
       await GlobalExceptionHandler.getInstance()
```
Comment (VJag) [inline at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart:316]: Let's add some logs here.

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart [hunk 3]
```diff
@@ -310,10 +310,10 @@ class AtSecondaryServerImpl implements AtSecondaryServer {
...
           .handle(e, atConnection: connection);
-    } on Error catch (e, trace) {
+    } on Error catch (e) {
       logger.severe(e.toString());
```
Comment (VJag) [inline at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart]: Just thinking. Does adding some additional context to the logs like 'Exception in executing verb etc...'

File: at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart [hunk 4]
```diff
@@ -39,70 +39,75 @@ class StatsNotificationService {
   }
 
   final _logger = AtSignLogger('StatsNotificationService');
-  final String _currentAtSign =
-      AtSecondaryServerImpl.getInstance().currentAtSign;
-  AtCommitLog? _atCommitLog;
-  DateTime _lastSentTime = DateTime.now().toUtc();
+  late String currentAtSign;
+  AtCommitLog? atCommitLog;
+  List<InboundConnection>? connectionsList;
```
Comment (VJag) [inline at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart]: I believe this for the unit tests. Should we keep it the way it was and mock "InboundConnectionPool.getInstance().getConnections();" and return the connections you need for the testing.

Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart]: Tried the approach of mocking InboundConnectionPool. It is tricky to mock InboundConnections.

File: at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/lib/src/notification/stats_notification_service.dart]: Made the connectionsList global variable to pass mock connections for unit test

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart:316]: Sure, will added logs.

File: at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart (hunk 3 above)
Comment (sitaram-kalluri) [reply at_secondary/at_secondary_server/lib/src/server/at_secondary_impl.dart]: Sure, will add the logs.

**Review by murali-shris — APPROVED** [review]

**Pull request merged**